lib/Runtime/Math/JavascriptMath.cpp

```diff
@@ -169,45 +169,38 @@ namespace Js
             Assert(aRight != nullptr);
             Assert(scriptContext != nullptr);
 
-            // If both sides are numbers, then we can do the addition directly, otherwise
-            // we need to call the helper.
-            if(JavascriptNumber::Is(aLeft))
+            Js::TypeId typeLeft = JavascriptOperators::GetTypeId(aLeft);
+            Js::TypeId typeRight = JavascriptOperators::GetTypeId(aRight);
+
+            if (typeRight == typeLeft)
             {
-                if(JavascriptNumber::Is(aRight))
+                // If both sides are numbers/string, then we can do the addition directly
+                if(typeLeft == TypeIds_Number)
                 {
                     double sum = JavascriptNumber::GetValue(aLeft) + JavascriptNumber::GetValue(aRight);
                     return JavascriptNumber::ToVarNoCheck(sum, scriptContext);
                 }
-                else if(TaggedInt::Is(aRight))
-                {
-                    double sum = TaggedInt::ToDouble(aRight) + JavascriptNumber::GetValue(aLeft);
-                    return JavascriptNumber::ToVarNoCheck(sum, scriptContext);
-                }
-            }
-            else if(JavascriptNumber::Is(aRight))
-            {
-                if(TaggedInt::Is(aLeft))
-                {
-                    double sum = TaggedInt::ToDouble(aLeft) + JavascriptNumber::GetValue(aRight);
-                    return JavascriptNumber::ToVarNoCheck(sum, scriptContext);
-                }
-            }
-            else if(TaggedInt::Is(aLeft))
-            {
-                if(TaggedInt::Is(aRight))
+                else if (typeLeft == TypeIds_Integer)
                 {
                     __int64 sum = TaggedInt::ToInt64(aLeft) + TaggedInt::ToInt64(aRight);
                     return JavascriptNumber::ToVar(sum, scriptContext);
                 }
+                else if (typeLeft == TypeIds_String)
+                {
+                    return JavascriptString::Concat(JavascriptString::FromVar(aLeft), JavascriptString::FromVar(aRight));
+                }
             }
-            else if (TaggedInt::Is(aRight))
+            else if(typeLeft == TypeIds_Number && typeRight == TypeIds_Integer)
```

Comment:
Super nit: in some cases, you have a space in between the if and (, and in some cases you don't- can you pick one and stick to it 😄?